azimuth/types/tag.py

```diff
@@ -56,6 +56,26 @@ class SmartTag(str, Enum):
     no_smart_tag = "NO_SMART_TAGS"
 
 
+class SmartTagFamily(str, Enum):
+    extreme_length = "extreme_length"
+    partial_syntax = "partial_syntax"
+    dissimilar = "dissimilar"
+    almost_correct = "almost_correct"
+    behavioral_testing = "behavioral_testing"
+    uncertain = "uncertain"
+
+
+SMART_TAG_FAMILY_DATASET = [
+    SmartTagFamily.extreme_length,
+    SmartTagFamily.partial_syntax,
+    SmartTagFamily.dissimilar,
+]
+SMART_TAG_FAMILY_PIPELINE = [
```

Comment:
Any objection to using plural "families" and putting dataset/pipeline as an adjective at the beginning, similarly to the types like `ValuePerPipelineSmartTag`?
```suggestion
DATASET_SMART_TAG_FAMILIES = [
    SmartTagFamily.extreme_length,
    SmartTagFamily.partial_syntax,
    SmartTagFamily.dissimilar,
]
PIPELINE_SMART_TAG_FAMILIES = [
```